asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -317,8 +320,41 @@ def get_sync_data(self, context, router_ids=None, active=None, host=None):
             router["rt_export"] = list(set(rt_export))
             router["rt_import"] = list(set(rt_import))
 
+            all_ports = [x["id"] for x in router.get("_interfaces", [])]
+            if gw_port:
+                all_ports.append(gw_port["id"])
+
+            router["fwaas_policies"] = self.get_fwaas_policies(context, all_ports)
+
         return routers
 
+    def get_fwaas_policies(self, context, port_ids):
+        policies = {}
+
+        for port_id in port_ids:
+            with context.session.begin(subtransactions=True):
```

Comment:
It doesn't really hurt to use an individual transaction, yet since this is called out of the context of a router, I think it makes sense to let the tx span multiple ports.